xmodule/library_tools.py

```diff
@@ -155,48 +187,66 @@ def update_children(self, dest_block, user_perms=None, version=None):
             return
 
         source_blocks = []
+
         library_key = dest_block.source_library_key
-        if version:
+        is_v2_lib = isinstance(library_key, LibraryLocatorV2)
+
+        if version and not is_v2_lib:
             library_key = library_key.replace(branch=ModuleStoreEnum.BranchName.library, version_guid=version)
-        library = self._get_library(library_key)
+
+        library = self._get_library(library_key, is_v2_lib)
+
         if library is None:
             raise ValueError(f"Requested library {library_key} not found.")
         if user_perms and not user_perms.can_read(library_key):
             raise PermissionDenied()
-        filter_children = (dest_block.capa_type != ANY_CAPA_TYPE_VALUE)
-        if filter_children:
-            # Apply simple filtering based on CAPA problem types:
-            source_blocks.extend(self._problem_type_filter(library, dest_block.capa_type))
-        else:
-            source_blocks.extend(library.children)
-
-        with self.store.bulk_operations(dest_block.location.course_key):
-            dest_block.source_library_version = str(library.location.library_key.version_guid)
-            self.store.update_item(dest_block, self.user_id)
-            head_validation = not version
-            dest_block.children = self.store.copy_from_template(
-                source_blocks, dest_block.location, self.user_id, head_validation=head_validation
-            )
-            # ^-- copy_from_template updates the children in the DB
-            # but we must also set .children here to avoid overwriting the DB again
+
+        if not is_v2_lib:
+            if user_perms and not user_perms.can_read(library_key):
+                raise PermissionDenied()
```

Comment:
As @bradenmacdonald points out, the second half of this block (everything under `if not is_v2_lib`) is redundant with the first half of the block.